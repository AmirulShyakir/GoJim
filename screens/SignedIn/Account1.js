import EventTypeCard from "../../components/containers/EventTypeCard";
import CurrentPastFavouriteCard from "../../components/containers/CurrentPastFavouriteCard";
import CurrentPastFavouriteContainer from "../../components/containers/CurrentPastFavouriteContainer";

const Account1 = ({ navigation, route }) => {

  function handlePress(selected) {
    console.log("this is getting passed to the next page: " + selected)
    if (selected == "Favourite Facilities") {
      console.log("Navigating to Facility")
      //navigation.navigate("Facility", {key: selected});
    } else {
      //navigate to a page containing list of bookings
      navigation.navigate("MyBookings", selected);
    }
  }

  return (
    <CurrentPastFavouriteContainer>
      <CurrentPastFavouriteCard
        onPress={() => {
          handlePress("joe");
        }}
        image={{
          uri: "https://images.unsplash.com/photo-1522202176988-66273c2fd55f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80",
        }}
      >
        Upcoming Bookings
      </CurrentPastFavouriteCard>

      <CurrentPastFavouriteCard
        onPress={() => {
          handlePress("mama");
        }}
        image={{
          uri: "https://images.unsplash.com/photo-1510531704581-5b2870972060?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1173&q=80",
        }}
      >
        Past Bookings
      </CurrentPastFavouriteCard>
      <CurrentPastFavouriteCard
        onPress={() => {
          handlePress("Favourite Facilities");
        }}
        image={{
          uri: "https://firebasestorage.googleapis.com/v0/b/gojim-1d589.appspot.com/o/USC_Olympic-pool-1-1024x706.jpg?alt=media&token=0be3f293-bb9e-4119-a16a-901201713612",
        }}
      >
        Favourite Facilities
      </CurrentPastFavouriteCard>
    </CurrentPastFavouriteContainer>
  );
};

export default Account1;
