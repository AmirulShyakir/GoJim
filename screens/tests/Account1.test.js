import React from "react";
import renderer from "react-test-renderer";
import Account1 from "../SignedIn/Account1";
import { render } from "@testing-library/react-native";

test("Account1 Snapshot test", () => {
  const tree = renderer.create(<Account1 />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("it renders all cards correctly", () => {
  const { getByTestId } = render(<Account1 />);
  getByTestId("upcoming");
  getByTestId("past");
  getByTestId("favourites");
});
