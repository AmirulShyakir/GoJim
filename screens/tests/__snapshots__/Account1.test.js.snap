// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account1 Snapshot test 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#413636",
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
      "padding": 10,
      "paddingHorizontal": 15,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "30%",
        "margin": 7,
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="upcoming"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1522202176988-66273c2fd55f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Upcoming Bookings
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "30%",
        "margin": 7,
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="past"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1510531704581-5b2870972060?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1173&q=80",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Past Bookings
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "30%",
        "margin": 7,
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="favourites"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://firebasestorage.googleapis.com/v0/b/gojim-1d589.appspot.com/o/USC_Olympic-pool-1-1024x706.jpg?alt=media&token=0be3f293-bb9e-4119-a16a-901201713612",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Favourite Facilities
      </Text>
    </View>
  </View>
</View>
`;
